backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+—------------------------------------------------------------------------------------------------------------------
+***
+Test#4 - 261 - 309
+get data from userAnswer
+***
+
+***
+pre-req: 
+    !isSuperUser($userid) && isset($_SESSION["submission-$courseid-$coursevers-$duggaid-$moment"]) == TRUE &&
+    isset($_SESSION["submission-password-$courseid-$coursevers-$duggaid-$moment"] == TRUE
+***
+
+$pre-values = {
+        $hash = “AcIj-Ld6”
+        $hashpwd = “OjOEh8R”
+        $variant = 13
+    }
+
+    //Get the selected user by hash
+    SELECT password,timesSubmitted,timesAccessed,grade from userAnswer WHERE hash=$hash;
+
+***
+send parameters input to showDuggaservice.php
+***
+send 
+    {
+        $opt = “SAVDU”
+        $hash = “AcIj-Ld6”
+    }
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #5 - 279 - 287
+update submitted dugga
+
+***
+Pre-req: 
+    !isset($grade) || !$grade > 1 &&
+    isset($dbpwd) && strcmp($hashpwd,$dbpwd) === 0
+***
+
+***
+Pre-req variables
+***
+$pre-values =
+    {
+        $aid = 999
+        $cid = 2
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $moment = 2002
+        $answer = NULL
+    }
+
+    SQL-query: INSERT INTO userAnswer(aid,cid,hash,password,moment,useranswer) VALUES (pre-values[0], pre-values[1], pre-values[2], pre-values[3], pre-values[4], pre-values[5]);
+
+* Get newly created data
+    SELECT * WHERE hash=pre-values[2];
+
+***
+* Update information on userAnswer
+* send parameters input to showDuggaservice.php
+***
+send
+    {
+        $opt = SAVDU
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $answer = “im updated”
+    }
+
+    SQL-query: "UPDATE userAnswer SET submitted=NOW(), useranswer=$answer, timesSubmitted=timesSubmitted+1 WHERE hash=$hash AND password=$hashpwd
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = pre-values[2];
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #6 - 293 - 307
+submit dugga
+
+***
+Pre-req: !isset($grade) || !$grade > 1
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $opt = “SAVDU”
+        $cid = 2
```

Comment:
there is no course with 
$cid = 2
$coursevers = 52432

![image](https://user-images.githubusercontent.com/102600391/234771088-e4ecc9fb-603c-448f-b483-4c719f5a2877.png)

An insert needs to be made beforehand

Reply:
They are not dependent on each other, since vers is an attribute itself in userAnswer. however, cid must still be checked and will be added beforehand.